rust/operator-binary/src/discovery.rs

```diff
@@ -68,21 +67,24 @@ pub async fn build_discovery_configmaps(
             owner,
             resolved_product_image,
             &name,
-            service_hosts(svc, port_name)?,
+            listener_hosts(listeners, port_name)?,
         )?,
+        // backwards compat: nodeport service is now the same as the main service, access type
+        // is determined by the listenerclass.
+        // do we want to deprecate/remove this?
```

Comment:
Long term I'd agree. Short term the old name is part of our public API, so anyone relying on it already would have to reconfigure.

Reply:
I would prefer to make a clean cut, the listener-migration is breaking in any case - so users already need to modify stuff. I'm not aware of anyone using this and I would try to avoid having it for longer so that people don't start using it.

Reply:
Ok, let's leave the ConfigMap in just to get this PR merged. We can remove it later, we at least have a comment at the code :+1: